extension/src/editor/AspireEditorCommandProvider.ts

```diff
@@ -55,7 +55,19 @@ export class AspireEditorCommandProvider implements vscode.Disposable {
     public async processDocument(document: vscode.TextDocument): Promise<void> {
         const fileExtension = path.extname(document.uri.fsPath).toLowerCase();
         const isSupportedFile = getResourceDebuggerExtensions().some(extension => extension.getSupportedFileTypes().includes(fileExtension));
+
         vscode.commands.executeCommand('setContext', 'aspire.editorSupportsRunDebug', isSupportedFile);
+
+        if (this.isAppHostCsFile(document.uri.fsPath)) {
+            vscode.commands.executeCommand('setContext', 'aspire.fileIsAppHostCs', true);
+        }
+        else {
+            vscode.commands.executeCommand('setContext', 'aspire.fileIsAppHostCs', false);
+        }
+    }
+
+    private isAppHostCsFile(filePath: string): boolean {
+        return path.basename(filePath).toLowerCase() === 'apphost.cs';
```

Comment:
This doesn't seem like the best way to detect. Are we able to look at the file and see if it has a `#:sdk Aspire.AppHost.Sdk` in it?